charts/cheetah-application/values.schema.json

```diff
@@ -0,0 +1,462 @@
+      "type": "object",
+      "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.28.0/probe.json"
+    },
+    "livenessProbe": {
+      "description": "Whether to enable a liveness probe for the application. See https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/",
+      "type": "object",
+      "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.28.0/probe.json"
+    },
+    "readinessProbe": {
+      "description": "Whether to enable a readiness probe for the application. See https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/",
+      "type": "object",
+      "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.28.0/probe.json"
+    },
+    "resources": {
+      "description": "Resource limits. See [values.yaml](values.yaml) for formatting",
+      "type": "object",
+      "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.28.0/_definitions.json#/definitions/io.k8s.api.core.v1.ResourceRequirements"
+    },
+    "podSecurityContext": {
+      "description": "Security context for the entire pod. @default -- (see [values.yaml](values.yaml))",
+      "type": "object",
+      "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.28.0/podsecuritycontext.json"
+    },
+    "securityContext": {
+      "description": "Security context for the primary container. @default -- (see [values.yaml](values.yaml))",
+      "type": "object",
+      "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.28.0/securitycontext.json"
+    },
+    "monitoring": {
+      "type": "object",
+      "allOf": [
+        {
+          "if": {
+            "properties": {
+              "enabled": {
+                "const": true
+              }
+            }
+          },
+          "required": ["port", "path"]
+        }
+      ],
+      "properties": {
+        "enabled": {
+          "description": "Whether to enable Prometheus scraping by creating a ServiceMonitor resource",
+          "type": "boolean"
+        },
+        "port": {
+          "description": "Which port to look for Prometheus metrics",
+          "type": "number"
+        },
+        "path": {
+          "description": "Which path to look for Prometheus metrics",
+          "type": "string"
+        }
+      }
+    },
+    "pdb": {
+      "type": "object",
+      "allOf": [
+        {
+          "if": {
+            "properties": {
+              "create": {
+                "const": true
+              }
+            }
+          },
+          "then": {
+            "required": ["minAvailable", "maxUnavailable"]
```

Comment:
One and only one of these are required. Both can never be set

Reply:
I've made them mutually exclusive using the `oneOf` property https://github.com/trifork/cheetah-charts/pull/95/files#diff-2a53a173f70809a5260cef68a12d6e1fc4038a4345ee24fb26d0b712de32b7e4R345